tests/test_facedetector.py

```diff
@@ -386,3 +391,48 @@ def test_match_detection_one_image(self):
                             assert face.boundingBox.asDict() == detectOne.boundingBox.asDict()
                             assert face.landmarks5.asDict() == detectOne.landmarks5.asDict()
                             assert face.landmarks68.asDict() == detectOne.landmarks68.asDict()
+
+    def test_detect_one_large_image(self):
+        """
+        Test detection of one face from large image
+        """
+        for detector in self.detectors:
+            with self.subTest(detectorType=detector.detectorType):
+                detection = detector.detectOne(image=VLIMAGE_LARGRE)
+                self.assertFaceDetection(detection, VLIMAGE_LARGRE)
+
+    def test_detect_one_by_large_area(self):
+        """
+        Test detection of one face by large area
+        """
+        for detector in self.detectors:
+            with self.subTest(detectorType=detector.detectorType):
+                detection = detector.detectOne(image=VLIMAGE_LARGRE, detectArea=LARGE_AREA)
+                self.assertFaceDetection(detection, VLIMAGE_LARGRE)
+
+    def test_detect_one_by_small_area(self):
+        """
+        Test detection of one face by small area
+        """
+        for detector in self.detectors:
+            with self.subTest(detectorType=detector.detectorType):
+                detection = detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=SMALL_AREA)
+                self.assertIsNone(detection)
+
+    def test_detect_one_by_area_larger_image(self):
+        """
+        Test detection of one face by small area
+        """
+        for detector in self.detectors:
+            with self.subTest(detectorType=detector.detectorType):
+                with self.assertRaises(expected_exception=LunaSDKException):
+                    detection = detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=AREA_LARGER_IMAGE)
+
+    def test_detect_one_by_area_outside_image(self):
+        """
+        Test detection of one face by small area
+        """
+        for detector in self.detectors:
+            with self.subTest(detectorType=detector.detectorType):
+                with self.assertRaises(expected_exception=LunaSDKException):
+                    detection = detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=AREA_OUTSIDE_IMAGE)
```

Comment:
```
Need test for a face dtetection which does not match with images:

small detection rect(2 pixel)
big detection rect (10**4 pixels)
small image
the rectangle extends beyond the border of the image (with correct detection and not correct)
rectangle outside the border of the image
```
It was about of the mask estimation by image + detection

Reply:
I added tests to user input detection, but this usage is wrong

Reply:
no case when rect much larger than the image (width  = 10**5,  height = 10 **5)
no case when whole rect outside image (x= self.imageMedicalMask.rect.x + 10, y= self.imageMedicalMask.rect.y + 10, ...)

Reply:
This tests should be in facedetection tests. Users don't have to define the FaceDetection object themselves.

Reply:
user don't define  the FaceDetection object themselves. but if he detects two faces (F1 and F2) on two images (I1 and I2)  and  run mask estimation on image I1 and detection F2   we will have an error. We (sdk) should correct hamdle this case. If sdk handles not correct (Internal error for exanple) need create issue in jira.